Content.Server/_ES/Masks/Traitor/ESAddMaskOnUseSystem.cs

```diff
@@ -0,0 +1,104 @@
+﻿using Content.Server._ES.Masks.Traitor.Components;
+using Content.Server.Database.Migrations.Sqlite;
+using Content.Shared._ES.Masks.Components;
+using Content.Shared._ES.Masks.Traitor.Events;
+using Content.Shared.Damage.Systems;
+using Content.Shared.DoAfter;
+using Content.Shared.EntityEffects;
+using Content.Shared.Interaction;
+using Content.Shared.Mind;
+using Content.Shared.Mindshield.Components;
+using Content.Shared.Mobs.Systems;
+using Content.Shared.Popups;
+using Microsoft.CodeAnalysis;
+using Robust.Shared.Player;
+
+namespace Content.Server._ES.Masks.Traitor;
+
+public sealed class ESAddMaskOnUseSystem : EntitySystem
+{
+    [Dependency] private readonly ESMaskSystem _mask = default!;
+    [Dependency] private readonly SharedDoAfterSystem _doafter = default!;
+    [Dependency] private readonly MobStateSystem _mobState = default!;
+    [Dependency] private readonly SharedPopupSystem _popup = default!;
+    [Dependency] private readonly SharedMindSystem _mind = default!;
+    [Dependency] private readonly DamageableSystem _damageableSystem = default!;
+
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<ESAddMaskOnUseComponent, AfterInteractEvent>(OnInteract);
+        SubscribeLocalEvent<ESAddMaskOnUseComponent, ESAddMaskOnUseDoAfterEvent>(OnDoAfter);
+    }
+
+    private void OnInteract(EntityUid uid, ESAddMaskOnUseComponent component, ref AfterInteractEvent args)
+    {
+        if (args.Target == null)
+            return;
+
+        if (!_mobState.IsCritical((EntityUid)args.Target) && component.RequireCrit)
+            return;
+
+        if (HasComp<MindShieldComponent>(args.Target) && component.MindshieldPrevent)
+            return;
+
+        if (!_mind.TryGetMind((EntityUid)args.Target!, out var mind, out var mindComponent)) // No SSD people
+            return;
+
+        _mask.TryGetTroupe((mind, mindComponent), out var troupe);
+
+        if (troupe == component.TroupeToAdd)
+            return;
+
+        if (component.Used)
+        {
+            _popup.PopupEntity(Loc.GetString("subverter-chip-used"), args.User, args.User, PopupType.Medium);
+            return;
+        }
+
+        var DoAfterArgs = new DoAfterArgs(EntityManager, args.User, component.Delay, new ESAddMaskOnUseDoAfterEvent(), eventTarget: uid, args.Target, used: uid)
+        {
+            BlockDuplicate = true,
+            BreakOnMove = true,
+            BreakOnDamage = true,
+            BreakOnDropItem = true,
+            NeedHand = true,
+            MovementThreshold = 0.5f
+        };
+
+        _doafter.TryStartDoAfter(DoAfterArgs);
+
+        _popup.PopupEntity(Loc.GetString("subverter-chip-implanting"), uid, PopupType.MediumCaution);
+
+    }
+    private void OnDoAfter(EntityUid uid, ESAddMaskOnUseComponent component, ESAddMaskOnUseDoAfterEvent args)
+    {
+        if (args.Cancelled || args.Handled)
+            return;
+
+        var target = (EntityUid)args.Target!;
```

Comment:
```suggestion
        if (args.Target is not { } target)
            return;
```
In general you should always handle nullable entities and avoid casting away nullability without checking properly (that's what the `!` and `(EntityUid)` cast are doing)

this code works properly in the event that the Target is unexpectedly null and also gives you the same temporary variable.